File: ios/Hyperlinks/Hyperlinks/ViewController.m
//
//  ViewController.m
//  Hyperlinks
//
//  Created by 村上 幸雄 on 12/05/03.
//  Copyright (c) 2012年 ビッツ有限会社. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@end

@implementation ViewController

@synthesize textView = _textView;

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
    
    self.textView.editable = NO;
    self.textView.dataDetectorTypes = UIDataDetectorTypeLink;
    self.textView.text = @"This is a demonstration.\nhttp://www.bitz.co.jp/\nThank you.";
}

- (void)viewDidUnload
{
    self.textView = nil;
    [super viewDidUnload];
    // Release any retained subviews of the main view.
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    return (interfaceOrientation != UIInterfaceOrientationPortraitUpsideDown);
}

@end

File: mac/MorphologicalAnalysis/MorphologicalAnalysis/ViewController.swift
//
//  ViewController.swift
//  MorphologicalAnalysis
//
//  Created by 村上幸雄 on 2019/04/14.
//  Copyright © 2019 Bitz Co., Ltd. All rights reserved.
//

import Cocoa

class ViewController: NSViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        analyzeNaturalLanguageText()
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }
    
    func analyzeNaturalLanguageText() {
        let text = "東京は、哀しい活気を呈していた、とさいしょの書き出しの一行に書きしるすというような事になるのではあるまいか、と思って東京に舞い戻って来たのに、私の眼には、何の事も無い相変らずの「東京生活」のごとくに映った。"
        let linguisticTagger = NSLinguisticTagger(tagSchemes: NSLinguisticTagger.availableTagSchemes(forLanguage: "ja"), options: 0)
        linguisticTagger.string = text
        linguisticTagger.enumerateTags(in: NSRange(location: 0, length: text.count),
                             scheme: NSLinguisticTagScheme.tokenType,
                             options: [.omitWhitespace]) {
                                tag, tokenRange, sentenceRange, stop in
                                let subString = (text as NSString).substring(with: tokenRange)
                                print("\(subString) : \(String(describing: tag))")
        }
    }
}

